packages/test/test-end-to-end-tests/src/test/createNewSummaryCachingTests.spec.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { MockLogger } from "@fluidframework/test-runtime-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeOptions } from "@fluidframework/container-runtime";
+import { flattenRuntimeOptions } from "./flattenRuntimeOptions";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+    public get _context() {
+        return this.context;
+    }
+}
+
+describeNoCompat("Cache CreateNewSummary", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const IdleDetectionTime = 100;
+    const summaryConfigOverrides: Partial<ISummaryConfiguration> = {
+        idleTime: IdleDetectionTime,
+        maxTime: IdleDetectionTime * 12,
+    };
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            generateSummaries: true,
+            initialSummarizerDelayMs: 10,
+            summaryConfigOverrides,
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        undefined,
+        flattenRuntimeOptions(runtimeOptions),
+    );
+
+    let mockLogger: MockLogger;
+
+    beforeEach(function() {
+        provider = getTestObjectProvider();
+        // Currently, only ODSP caches new summary.
+        if (provider.driver.type !== "odsp") {
+            this.skip();
+        }
+    });
+
+    it("should fetch from cache when second client loads the container", async () => {
+        mockLogger = new MockLogger();
+        // Create a Container for the first client.
+        const mainContainer = await provider.createContainer(runtimeFactory, { logger: mockLogger });
+        assert.strictEqual(mainContainer.attachState, "Attached", "container was not attached");
+        const mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await dataObjectFactory.createInstance(mainDataStore._context.containerRuntime);
+
```

Comment:
You should add an `await provider.ensureSynchronized()` here to ensure that the new data store creation is sync'd before you load a second container.